### DISCUSSION TITLE
How to check if the current process is running Tauri (not web only)?

### DISCUSSION THREAD

**Discussion by boan-anbo** [tauri-apps/tauri#2725] [Q&A]
Is there an equivalent to `is-electron` or other native means to check if it's current running in Tauri?

**Comment by FabianLars** [comment]
you can check for the `__TAURI__` object: `if (window.__TAURI__) {}`

**Reply by boan-anbo** [reply]
Cool thanks!

**Reply by TimerErTim** [reply]
The window object is cleared upon page changes. So the `__TAURI__` variable is not retained and not a reliable source for determining if inside a Tauri app.

Is there another way to check?

**Reply by simonhyll** [reply]
Depends on how advanced you want to make it. Personally I use an approach that relies on compiling a WASM library differently depending on whether or not I'm compiling for Tauri support. The most reliable solution is some manner of conditional compilation of your code, whether you do it with Webpack, Vite or WASM is entirely up to you. Checking for `__TAURI__` however on the window object is probably the easiest solution however, possibly waiting to run any commands that check for it until the page is fully loaded just to be sure.

**Reply by joepio** [reply]
~~I think that `window.__TAURI__` is only available when `build.withGlobalTauri` is enabled, since a release or two. So it might be better to use `window.__TAURI_METADATA__`, which should always be available.~~

Incorrect, may be a bug.

**Reply by FabianLars** [reply]
> I think that window.__TAURI__ is only available when build.withGlobalTauri is enabled, since a release or two.

Nope, not the case. I think what you're seeing here is that you're navigating/redirecting to external urls? In that case only `__TAURI_METADATA__` and `__TAURI_POST_MESSAGE__` will be left, _which is a bug_! Sooner or later (as-in, in some future release) these 2 won't be available after redirect anymore either.

**Reply by joepio** [reply]
> I think what you're seeing here is that you're navigating/redirecting to external urls? 

I'm not doing that in my app, so that should not be the case. So I guess what I'm seeing is a bug then! Anyways, for me checking `window.__TAURI_METADATA__` works at this moment, and `window.__TAURI__` does not.

**Reply by akiross** [reply]
Is this answer still valid for v2?

Edit: apparently not.

**Comment by akiross** [comment]
In v2, it seems from [this answer](https://github.com/tauri-apps/tauri/discussions/6119#discussioncomment-8845935) that `window.__TAURI_INTERNALS__` might be set in tauri.

**Comment by unlocomqx** [comment]
(v2) [also works](https://v2.tauri.app/reference/javascript/api/namespacecore/#istauri)
```ts
import {isTauri} from "@tauri-apps/api/core"
```
